### DISCUSSION TITLE
Inline shortcuts are sometimes not handled correctly

### DISCUSSION THREAD

**Discussion by GiovanniSalmeri** [datenstrom/community#806] [Report a bug] [answered]
Hello. I stumbled upon a little problem with the inline shortcuts. When there is one inline shortcut at the start of a line and one at the end, the whole line is interpreted as a block shortcut. Obviously the result is not the expected one. For example:

```
[youtube K7g-VeD2kkU] Possibly some text... [youtube EmZF3kBZQ6E]
```

It seems to me that the cause of this is in markdown.php:3878:

```
$text = preg_replace_callback("/^\s*\[(\w+)(.*?)\]\s*$/", array($this, "_doAutoLinks_shortcutBlock_callback"), $text);
```

This checks only if there is a `[` at the start of the line and a `]` ath the end. This has a positive side-effect of allowing brackets inside a block shortcut, for example: `[foobar "This is a text with [some] brackets"]`, but the negative effect (above described) seems to me worse (especially since brackets are anyway *not* allowed inside inline shortcuts). After some tests, it seems to me that the problem could easily be solved with this little modification:

```
$text = preg_replace_callback("/^\s*\[(\w+)([^]]*)\]\s*$/", array($this, "_doAutoLinks_shortcutBlock_callback"), $text);
```

Or am I missing something? Thank you very much!

**Comment by markseuffert** [comment]
I can confirm the bug, thanks for the detailed information. In theory the regex uses a non-greedy matching, but in practice it sometimes doesn't work as expected. By the way, while testing I found out that only extensions are affected that handle both `block` type and `inline` type shortcuts, for example the image extension is not affected.

Here is another example for reproducing the problem:

```
[image photo.jpg] [image photo.jpg] [image photo.jpg]

[youtube fhs55HEl-Gc] [youtube fhs55HEl-Gc] [youtube fhs55HEl-Gc]
```

**Reply by GiovanniSalmeri** [reply]
> In theory the regex uses a non-greedy matching

I reflected on this and I guess that in this case the non-greedy matching is voided by the `$` anchor: "as few characters as possible... but anyway until a bracket at the end of the string". If I understand well the mysterious mind of regex engines 😄

**Comment by markseuffert** [comment, marked as answer]
Thanks for your bug report. I updated the [Markdown extension](https://github.com/annaesvensson/yellow-markdown) and the [Parsedown extension](https://github.com/annaesvensson/yellow-parsedown). There was an error with detecting block type shortcuts, a false positive, which prevented detecting inline type shortcuts.

Should be fixed now, please try it out.

**Reply by GiovanniSalmeri** [reply]
Perfect, thank you!
